flink-core/src/main/java/org/apache/flink/configuration/HeartbeatManagerOptions.java

```diff
@@ -32,15 +34,40 @@
     public static final ConfigOption<Long> HEARTBEAT_INTERVAL =
             key("heartbeat.interval")
                     .defaultValue(10000L)
-                    .withDescription("Time interval for requesting heartbeat from sender side.");
+                    .withDescription(
+                            "Time interval between heartbeat RPC requests from the sender to the receiver side.");
 
     /** Timeout for requesting and receiving heartbeat for both sender and receiver sides. */
     @Documentation.Section(Documentation.Sections.EXPERT_FAULT_TOLERANCE)
     public static final ConfigOption<Long> HEARTBEAT_TIMEOUT =
             key("heartbeat.timeout")
                     .defaultValue(50000L)
                     .withDescription(
-                            "Timeout for requesting and receiving heartbeat for both sender and receiver sides.");
+                            "Timeout for requesting and receiving heartbeats for both sender and receiver sides.");
+
+    private static final String HEARTBEAT_RPC_FAILURE_THRESHOLD_KEY =
+            "heartbeat.rpc-failure-threshold";
+
+    @Documentation.Section(Documentation.Sections.EXPERT_FAULT_TOLERANCE)
+    public static final ConfigOption<Integer> HEARTBEAT_RPC_FAILURE_THRESHOLD =
+            key(HEARTBEAT_RPC_FAILURE_THRESHOLD_KEY)
+                    .intType()
+                    .defaultValue(2)
+                    .withDescription(
+                            Description.builder()
+                                    .text(
+                                            "The number of failed heartbeat RPCs until a heartbeat target is marked as unreachable. "
```

Comment:
I think we should mention that N RPCs have to fail in a row for this to trigger.

Reply:
Yes, this is a good point. I will update the description.